### ISSUE TITLE
Unexpected results with BLPattern

### ISSUE THREAD

**Issue by abu-irrational** [blend2d/blend2d#200]
Here's a stunning test showing an unexpected result with BLPattern.

Tested on Windows with blend2d 'master' branch dated 10-mar-2024.

see code below

What's more surprising is that this minimal code, when excuted with fiddle.blend2d.com,
  works correctly !

What I'm trying to do:

  On the main BL::context,
  draw a circle at the center of the image, filled with with the center of a large pattern.

 Here you can see 
* pattern.png  (300x300)
![pattern](https://github.com/blend2d/blend2d/assets/13416847/775ce386-d7b1-4ccd-bc7f-69effe866760)
* the expected result.
![ctx-ok](https://github.com/blend2d/blend2d/assets/13416847/bb682c47-bfe5-4984-8bd5-379ef4ae9d90)
* the (wrong) result
![ctx-bad](https://github.com/blend2d/blend2d/assets/13416847/f61cfeff-7a23-42ff-b260-34f1c1388c04)

I've tried many tricks and variations ...
  What's stunning is that, if I just add a small displacement (0.01) to the 'center'
  of the pattern, I get the correct result
  BUT if this displacement is 0.001 , then the result is still wrong.

  Then I introduced another variation:
   if I set
     myPattern.setExtendMode(BL_EXTEND_MODE_REFLECT)
  then the result is correct, but I don't understand why, since
  I "fetch" just the basic pattern-image, no need to fetch 'extended' parts ..

Please, can you confirm or reject this issue with the latest release ?

---- code ---- tesPattern.cpp

```
#include <blend2d.h>

int main(int argc, char* argv[]) {

   // create an image (3003x300) that will be used as a BLPattern
  BLImage imgPattern(300, 300, BL_FORMAT_PRGB32);
  double w = imgPattern.width();
  double h = imgPattern.height();
  double xc = w/2.0;
  double yc = h/2.0;
  {
	  BLContext ctxS(imgPattern);
	  ctxS.fillAll(BLRgba32(0xFFff0000));
	  ctxS.fillCircle(xc,yc ,0.21*w,  BLRgba32(0xFF0000ff));
	  ctxS.fillCircle(xc,yc, 0.11*w,  BLRgba32(0xFF888888));
	  ctxS.strokeLine(0,0,w,h,  BLRgba32(0xFFffff00));
	  ctxS.strokeLine(0,h,w,0,  BLRgba32(0xFFffffff));
	  ctxS.strokeLine(0,yc,w,yc,  BLRgba32(0xFFffffff));
	  ctxS.strokeLine(xc,0,xc,h,  BLRgba32(0xFFffffff));
	  ctxS.end();

		// just for check
  	  imgPattern.writeToFile("pattern.png");
  }

   // create the main BLContext
  BLImage img(400, 400, BL_FORMAT_PRGB32);
  BLContext ctx(img);
  ctx.clearAll();

  ctx.setTransform( {1,0,0,1, 200,200} ); //put the origin at the center of img

  BLPattern myPattern;
  myPattern.setImage(imgPattern);
    // put the origin of the pattern in its center
  myPattern.setTransform({1,0,0,1, xc, yc});

  // experimental variations :
  // myPattern.setTransform({1,0,0,1, xc+0.01, yc});  // it works !
  // myPattern.setTransform({1,0,0,1, xc+0.001, yc}); // still wrong
  // myPattern.setExtendMode(BL_EXTEND_MODE_REFLECT); // it works !
  ctx.setFillStyle(myPattern);

  ctx.fillCircle(0,0,90);
  ctx.end();

  img.writeToFile("ctx.png");

  return 0;
}

```-----------

**Comment by kobalicek** [comment]
I think this could be related to JIT pipeline generator not generating the intended code in some configurations, because when I put this to fiddle it shows the correct output:

```c++
#include <blend2d.h>

BLImage render(const BLContextCreateInfo& cci) {
  BLImage imgPattern(300, 300, BL_FORMAT_PRGB32);
  double w = imgPattern.width();
  double h = imgPattern.height();
  double xc = w/2.0;
  double yc = h/2.0;

  {
	  BLContext ctxS(imgPattern);
	  ctxS.fillAll(BLRgba32(0xFFff0000));
	  ctxS.fillCircle(xc,yc ,0.21*w,  BLRgba32(0xFF0000ff));
	  ctxS.fillCircle(xc,yc, 0.11*w,  BLRgba32(0xFF888888));
	  ctxS.strokeLine(0,0,w,h,  BLRgba32(0xFFffff00));
	  ctxS.strokeLine(0,h,w,0,  BLRgba32(0xFFffffff));
	  ctxS.strokeLine(0,yc,w,yc,  BLRgba32(0xFFffffff));
	  ctxS.strokeLine(xc,0,xc,h,  BLRgba32(0xFFffffff));
	  ctxS.end();
  }

   // create the main BLContext
  BLImage img(400, 400, BL_FORMAT_PRGB32);
  BLContext ctx(img);
  ctx.clearAll();

  ctx.setTransform( {1,0,0,1, 200,200} ); //put the origin at the center of img

  BLPattern myPattern;
  myPattern.setImage(imgPattern);
    // put the origin of the pattern in its center
  myPattern.setTransform({1,0,0,1, xc, yc});

  // experimental variations :
  // myPattern.setTransform({1,0,0,1, xc+0.01, yc});  // it works !
  // myPattern.setTransform({1,0,0,1, xc+0.001, yc}); // still wrong
  // myPattern.setExtendMode(BL_EXTEND_MODE_REFLECT); // it works !
  ctx.setFillStyle(myPattern);

  ctx.fillCircle(0,0,90);
  ctx.end();

  return img;
}
```

**Comment by kobalicek** [comment]
Marking this as a bug - this needs a test suite to be sure.

**Comment by kobalicek** [comment]
Indeed, this was a bug and it's now fixed in v0.11.0! Thanks for the report!
